lib/ecto/adapters/clickhouse/connection.ex

```diff
@@ -75,10 +75,10 @@ defmodule Ecto.Adapters.ClickHouse.Connection do
       group_by,
       having,
       window,
-      combinations,
       order_by,
       limit,
-      offset
+      offset,
+      combinations
```

Comment:
With that change the generated SQL is

```sql
SELECT p0."title" FROM "posts" AS p0
WHERE (NOT (p0."public"))
ORDER BY p0."counter"
LIMIT 1
UNION ALL (
  SELECT p0."title" FROM "posts" AS p0
  WHERE (p0."public")
  ORDER BY p0."counter"
  LIMIT 1
)
```

Reply:
I'll add a test with params to make sure they aren't dependant on this ordering.

Reply:
Ecto param handling indeed assumes unions and co (combinations) come before offset and limit, but that doesn't break our SQL since we use named params instead of `?`.

Reply:
Hm, but then we are working against Ecto assumptions in that Ecto wants to allow the user to order and limit the result set after applying `union`, but ClickHouse doesn't allow that. I guess we'll add it to a "caveats" section [like in `ch`](https://github.com/plausible/ch#caveats)

Reply:
oh wow, that was a fast fix! 

Interesting, I did not know about that, thanks for sharing! :)